### PR TITLE
document reloader dependency on actuator-autoconfigure

### PR DESCRIPTION
**Pull request by paulczar**
Signed-off-by: Paul Czarkowski <username.taken@gmail.com>

### CODE REVIEW

**Comment by paulczar** [comment]
The error in circleci appears to be unrelated to my change.

**Comment by geoand** [comment]
I don't think this is actually needed...

**Comment by spencergibb** [comment]
I think it is, the plain starter doesn't include config.

**Comment by geoand** [comment]
I'm looking to see if the reload example works as is

**Comment by paulczar** [comment]
It definitely needs the actuator and reload endpoint enabled, I was bashing my head against this all day yesterday.

File: README.adoc [hunk 1]
```diff
@@ -476,6 +476,15 @@ This means, for example, that using polling on file mounted secret sources does
 **Notes**:
 - Properties under *spring.cloud.kubernetes.reload.* should not be used in config maps or secrets: changing such properties at runtime may lead to unexpected results;
 - Deleting a property or the whole config map does not restore the original state of the beans when using the *refresh* level.
+- You need to have the reload actuator dependency in your pom file for reload and the reload actuator enabled `management.endpoint.restart.enabled=true` in your `bootstrap.yaml` or `application.yaml`:
+
+```xml
+<dependency>
```
Comment (ryanjbaxter) [inline README.adoc]: If this is needed we should specify the spring boot starter here

**Comment by geoand** [comment]
I just tried https://github.com/spring-cloud/spring-cloud-kubernetes/tree/master/spring-cloud-kubernetes-examples/kubernetes-reload-example which I just realized that is **does** have `spring-boot-actuator-autoconfigure`.

So yeah, this makes sense!

File: README.adoc (hunk 1 above)
Comment (geoand) [reply README.adoc]: We probably should include the dependency in `spring-cloud-starter-kubernetes-config`

File: README.adoc [hunk 2]
```diff
@@ -476,6 +476,15 @@ This means, for example, that using polling on file mounted secret sources does
 **Notes**:
 - Properties under *spring.cloud.kubernetes.reload.* should not be used in config maps or secrets: changing such properties at runtime may lead to unexpected results;
 - Deleting a property or the whole config map does not restore the original state of the beans when using the *refresh* level.
+- You need to have the reload actuator dependency in your pom file for reload and the reload actuator enabled `management.endpoint.restart.enabled=true` in your `bootstrap.yaml` or `application.yaml`:
```
Comment (ryanjbaxter) [inline README.adoc]: I dont think specifically we need to set `management.endpoint.restart.enabled=true`.  Maybe if you change to `spring.cloud.kubernetes.reload.strategy=restart_context` you might?  Is that what you did.  We should just need to enable the refresh endpoint by default.
```
management:
  endpoints:
    web:
      exposure:
        include: refresh
```

**Comment by geoand** [comment]
@paulczar Would you also please update the reload example with the changes proposed by @ryanjbaxter?

File: README.adoc (hunk 2 above)
Comment (paulczar) [reply README.adoc]: I've tried that.

```
spring.cloud.kubernetes.reload.strategy: refresh
#management.endpoint.restart.enabled: true
management.endpoints.web.exposure.include: refresh
```

gives me the following error during startup

```
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.cloud.context.restart.RestartEndpoint' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
```

**Comment by spencergibb** [comment]
Actuator needs to remain optional.

File: README.adoc (hunk 1 above)
Comment (paulczar) [reply README.adoc]: good catch, switched it across to use the starter

File: README.adoc (hunk 2 above)
Comment (ryanjbaxter) [reply README.adoc]: @paulczar did you include `spring-boot-starter-actuator`?  Also can you try
```
management:
  endpoint:
    restart:
      enabled: true
```

Comment (paulczar) [reply README.adoc]: yeah both those are what the PR is adding to the docs :)

**Pull request merged**